### PR TITLE
TO DELETE - Use Initializer in ERC20

### PR DESCRIPTION
**Pull request by leomassazza**

### CODE REVIEW

File: packages/core-modules/contracts/modules/OwnerModule.sol [hunk 1]
```diff
@@ -3,4 +3,7 @@ pragma solidity ^0.8.0;
 
 import "@synthetixio/core-contracts/contracts/ownership/Ownable.sol";
 
-contract OwnerModule is Ownable {}
```
Comment (leomassazza) [inline packages/core-modules/contracts/modules/OwnerModule.sol:6]: remove linter warning

**Review by ajsantander — APPROVED** [review]

File: packages/core-contracts/test/proxy/Beacon.test.js [hunk 2]
```diff
@@ -14,16 +14,20 @@ describe('Beacon', () => {
...
   describe('when a beacon is deployed', () => {
     before('deploy the beacon and set an initial implementation', async () => {
-      let factory;
+      let factory, tx;
 
       factory = await ethers.getContractFactory('ERC20');
-      ERC20Mock1 = await factory.deploy('Random Token 1', 'rnd1', 18);
-      ERC20Mock2 = await factory.deploy('Random Token 2', 'rnd2', 18);
+      ERC20Mock1 = await factory.deploy();
+      tx = await ERC20Mock1.initialize('Random Token 1', 'rnd1', 18);
```
Comment (i-stam) [inline packages/core-contracts/test/proxy/Beacon.test.js:21]: in theory you can even avoid initializing them! they are just used for testing

File: packages/core-contracts/contracts/token/ERC20.sol [hunk 3]
```diff
@@ -2,18 +2,22 @@
...
 
-    constructor(
+    function initialize(
         string memory tokenName,
         string memory tokenSymbol,
         uint8 tokenDecimals
-    ) {
+    ) public virtual {
         ERC20Namespace storage store = _erc20Storage();
+        if (bytes(store.name).length > 0 || bytes(store.symbol).length > 0 || store.decimals > 0) {
```
Comment (i-stam) [inline packages/core-contracts/contracts/token/ERC20.sol:18]: again in theory, this check can fail, even though it wouldn't be practical, my point is, should we use a logic like the initializer contract of OZ. Maybe for future implementations

**Review by i-stam — APPROVED** [review]
Just left a couple of comments. Not prohibiting for now, but we should keep a note.

**Pull request closed without merging**